### PR TITLE
passing data to TestGetSendsCorrectURL

### PR DESCRIPTION
**Pull request by vanjara**
Getting the test json data into TestGetSendsCorrectURL test function

### CODE REVIEW

File: weather_test.go [hunk 1]
```diff
@@ -35,7 +36,12 @@ func TestGetSendsCorrectURL(t *testing.T) {
 	t.Parallel()
 	var w weather.Conditions
 	s := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		fmt.Println("Inside the handler func")
+		data, err := ioutil.ReadFile("testdata/weather.json")
+		if err != nil {
+			fmt.Println("Error: ", err)
```
Comment (bitfield) [inline weather_test.go]: That's fine, but since this is a test, we could fail the test instead, couldn't we?

File: weather_test.go [hunk 2]
```diff
@@ -35,7 +36,12 @@ func TestGetSendsCorrectURL(t *testing.T) {
...
+			return
+		}
+		fmt.Fprintln(w, string(data))
```
Comment (bitfield) [inline weather_test.go]: Great! We could also have opened the file using `os.Open`, and then used `io.Copy` to copy the contents of the file (which is an `io.Reader`) to the ResponseWriter (which is an `io.Writer`). But this is totally fine too.

**Review by bitfield — COMMENTED** [review]
Ideal!

File: weather_test.go [hunk 3]
```diff
@@ -36,12 +35,12 @@ func TestGetSendsCorrectURL(t *testing.T) {
...
 	s := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		data, err := ioutil.ReadFile("testdata/weather.json")
+		file, err := os.Open("testdata/weather.json")
 		if err != nil {
-			fmt.Println("Error: ", err)
-			return
+			t.Fatal(err)
 		}
-		fmt.Fprintln(w, string(data))
+		defer file.Close()
```
Comment (vanjara) [inline weather_test.go:43]: @bitfield is this required? In the TestParseJson function, we have only done os.open but not done an explicit or a deferred file close.

Comment (bitfield) [reply weather_test.go:43]: It's always good practice to defer closing the resource.

**Pull request merged**